src/reynir_correct/pattern.py

```diff
@@ -1771,6 +2270,94 @@ def wrong_noun(nouns: Set[str], tree: SimpleTree) -> bool:
             )
         )
 
+        # 'af' incorrectly used with particular nouns
+        def wrong_noun_af(nouns: Set[str], tree: SimpleTree) -> bool:
+            """ Context matching function for the %noun macro in combination
+                with 'af' """
+            lemma = tree.own_lemma
+            if not lemma:
+                # The passed-in tree node is probably not a terminal
+                return False
+            try:
+                case = (set(tree.variants) & {"nf", "þf", "þgf", "ef"}).pop()
+            except KeyError:
+                return False
+            return (lemma + "_" + case) in nouns
+
+        NOUNS_AF: Set[str] = {
+            "beiðni_þgf",
+            "siður_þgf",
+            "tilefni_þgf",
+            "fyrirmynd_þgf"
+        }
+        # The macro %noun is resolved by calling the function wrong_noun_af()
+        # with the potentially matching tree node as an argument.
+        cls.ctx_noun_af = {"noun": partial(wrong_noun_af, NOUNS_AF)}
+        cls.add_pattern(
+            (
+                "af",  # Trigger lemma for this pattern
+                "PP > { P > { 'af' } NP > { %noun } }",
```

Comment:
Might want "PP > { P > { 'af' } NP > { .* %noun } }" to catch something like 'að þessu tilefni'. Need to look at the parse trees to get the exact pattern needed, however.

Reply:
Yes, same misunderstanding on my side.